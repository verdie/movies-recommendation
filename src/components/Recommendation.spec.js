import React from 'react'
import { shallow, mount } from 'enzyme'
import RecommendationWrapped, { Recommendation } from './Recommendation'
import toJson from 'enzyme-to-json'
import {Provider} from 'react-redux'
import store from '../store'

 const mockMovie = 
  {
    "popularity": 18.559,
    "vote_count": 1960,
    "video": false,
    "poster_path": "\/76KMEI1ZNU9HEZ4pmHTwgwa0u6d.jpg",
    "id": 153518,
    "adult": false,
    "backdrop_path": "\/k15COpeexV5Ad0tuLjBW2jQFoOz.jpg",
    "original_language": "en",
    "original_title": "The Angry Birds Movie",
    "genre_ids": [
      12, 16, 35
    ],
    "title": "The Angry Birds Movie",
    "vote_average": 6,
    "overview": "An island populated entirely by happy, flightless birds or almost entirely. In t" +
        "his paradise, Red, a bird with a temper problem, speedy Chuck, and the volatile " +
        "Bomb have always been outsiders. But when the island is visited by mysterious gr" +
        "een piggies, itâ€™s up to these unlikely outcasts to figure out what the pigs are " +
        "up to.",
    "release_date": "2016-05-20"
  }

describe('Recommendation component', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Provider store={store}><RecommendationWrapped current={mockMovie}/></Provider>)
    // const mApp = mount(<Provider store={store}><RecommendationWrapped/></Provider>)
  //  const app = wrapper.find(RecommendationWrapped)
  //  console.log('App length:', mApp.state())
  //  mApp.setState({
  //     current: mockMovie
  //   });
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})