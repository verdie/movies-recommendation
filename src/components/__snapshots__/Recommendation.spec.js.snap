// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recommendation component renders correctly 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <Connect(Recommendation)
    current={
      Object {
        "adult": false,
        "backdrop_path": "/k15COpeexV5Ad0tuLjBW2jQFoOz.jpg",
        "genre_ids": Array [
          12,
          16,
          35,
        ],
        "id": 153518,
        "original_language": "en",
        "original_title": "The Angry Birds Movie",
        "overview": "An island populated entirely by happy, flightless birds or almost entirely. In this paradise, Red, a bird with a temper problem, speedy Chuck, and the volatile Bomb have always been outsiders. But when the island is visited by mysterious green piggies, itâ€™s up to these unlikely outcasts to figure out what the pigs are up to.",
        "popularity": 18.559,
        "poster_path": "/76KMEI1ZNU9HEZ4pmHTwgwa0u6d.jpg",
        "release_date": "2016-05-20",
        "title": "The Angry Birds Movie",
        "video": false,
        "vote_average": 6,
        "vote_count": 1960,
      }
    }
  />
</ContextProvider>
`;
